tests/MongoDB.Driver.Tests/ListDatabasesTests.cs

```diff
@@ -0,0 +1,99 @@
+﻿/* Copyright 2020-present MongoDB Inc.
+*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*
+* http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*/
+
+using System;
+using FluentAssertions;
+using MongoDB.Bson;
+using MongoDB.Bson.TestHelpers.XunitExtensions;
+using MongoDB.Driver.Core.Misc;
+using MongoDB.Driver.Core.TestHelpers.XunitExtensions;
+using Xunit;
+
+namespace MongoDB.Driver.Tests
+{
+    public class ListDatabasesTests
+    {
+        private string _databaseName = $"authorizedDatabases{Guid.NewGuid()}";
+        private string _password = "authorizedDatabases";
+        private string _roleName = $"listDatabases{Guid.NewGuid()}";
+        private string _userName = $"authorizedDatabases{Guid.NewGuid()}";
+
+        [SkippableTheory]
+        [ParameterAttributeData]
+        public void Execute_should_return_the_expected_result_when_AuthorizedDatabases_is_used(
+            [Values(null, false, true)] bool? authorizedDatabases)
+        {
+            RequireServer.Check().Supports(Feature.ListDatabasesAuthorizedDatabases).Authentication(true);
+
+            var setupClient = DriverTestConfiguration.Client;
+            CreateListDatabasesRole(setupClient, _roleName);
+            CreateListDatabasesUser(setupClient, _userName, _password, _databaseName, _roleName);
+            setupClient.GetDatabase(_databaseName).GetCollection<BsonDocument>("test").InsertOne(new BsonDocument());
+
+            var settings = DriverTestConfiguration.Client.Settings.Clone();
+            settings.Credential = MongoCredential.FromComponents(mechanism: null, source: null, username: _userName, password: _password);
+            var testClient = new MongoClient(settings);
+
+            var options = new ListDatabasesOptions
+            {
+                AuthorizedDatabases = authorizedDatabases,
+                NameOnly = true,
+            };
+            var result = testClient.ListDatabases(options).ToList();
+
+            if (authorizedDatabases.HasValue && authorizedDatabases.Value)
+            {
+                result.Should().BeEquivalentTo(new BsonArray { new BsonDocument { { "name", _databaseName } } });
+            }
+            else
+            {
+                result.Count.Should().BeGreaterThan(1);
+            }
+        }
+
+        private void CreateListDatabasesRole(MongoClient client, string roleName)
```

Comment:
Should we test the case where the user doesn't have the role? e.g. verifying both branches as specified in the [docs here](https://docs.mongodb.com/manual/reference/command/listDatabases/#listdatabases-behavior)
> For MongoDB 4.0.6+, the listDatabases command returns different values based on the privileges assigned to the user who executes the command and the authorizedDatabases command option:
> If authorizedDatabases is unspecified, and If the user has listDatabases action on the cluster resource, listDatabases command returns all databases.
> If the user does not have listDatabases action on the cluster, listDatabases command returns only the databases for which the user has privileges (including databases for which the user has privileges on specific collections).
If authorizedDatabases is true, listDatabases command returns only the databases for which the user has privileges (including databases for which the user has privileges on specific collections).
If authorizedDatabases is false, and
If the user has listDatabases action on the cluster, listDatabases command returns all databases
If the user does not have listDatabases action on the cluster, listDatabases command errors with insufficient permissions.

I'm fine with not testing the case where the user doesn't have the role as IMHO that's more about the server's behavior than the driver's behavior, so feel free to disregard this!

Reply:
I would rather keep the test simple. Checking that what we send to the server has effect is enough IMO.